packages/shared/src/hooks/user/useUserShortByIdQuery.ts

```diff
@@ -0,0 +1,87 @@
+import { useQuery, UseQueryResult } from '@tanstack/react-query';
+import { USER_SHORT_BY_ID } from '../../graphql/users';
+import { generateQueryKey, RequestKey, StaleTime } from '../../lib/query';
+import { gqlClient } from '../../graphql/common';
+import type { PublicProfile } from '../../lib/user';
+import { useMutationSubscription } from '../mutationSubscription';
+import {
+  ContentPreferenceMutation,
+  contentPreferenceMutationMatcher,
+  mutationKeyToContentPreferenceStatusMap,
+} from '../contentPreference/types';
+import { PropsParameters } from '../../types';
+import {
+  ContentPreference,
+  ContentPreferenceType,
+} from '../../graphql/contentPreference';
+
+export type UseUserShortByIdQuery = UseQueryResult<PublicProfile>;
+export const useUserShortByIdQuery = ({
+  id,
+}: {
+  id: string;
+}): UseUserShortByIdQuery => {
+  const queryKey = generateQueryKey(RequestKey.UserShortById, null, { id });
+
+  const queryResult = useQuery({
+    queryKey,
+    queryFn: async (): Promise<PublicProfile> => {
+      const res = await gqlClient.request(USER_SHORT_BY_ID, { id });
+
+      return res.user;
+    },
+    staleTime: StaleTime.Default,
+    enabled: !!id,
+  });
```

Comment:
Can we disable retry when the error message is `user not found`?
![image](https://github.com/user-attachments/assets/5fe5c6c4-12a0-47df-aa48-472666e1ca68)

Reply:
Basically can have no retry even :)